invertase/react-native-firebase
# feat(app-check): add AppCheck implementation

### Description

Initial implementation of AppCheck module

What works:
 - `@react-native-firebase/app-check` package namespace to match firebase-js-sdk
 - `appCheck` javascript namespace to match firebase-js-sdk
 - multi-app support
 - debug provider on Android if app is debuggable
 - `setTokenAutoRefreshEnabled` method, and parameter on `activate`
 - `getToken` in e2e testing works!
 - docs

What needs implementing:
- firebase.json entry that toggles `FirebaseAppCheckTokenAutoRefreshEnabled` plist entry
- token refresh listeners on Android
- some mechanism to configure providers prior to `[FIRApp configure]` in AppDelegate.m. Maybe we need a startup hook?

I propose that plist entry is deferred as it can be toggled already by early adopters and it's an easy follow-on PR

I prose that refresh listeners on Android are deferred indefinitely, as they don't exist on iOS so would not be cross-platform anyway.

### Related issues

#5346 

https://github.com/wix/Detox/issues/2933

### Release Summary

Each commit is separate and works on it's own with a conventional commit message and comment

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- My change supports the following platforms;
- My change includes tests;
- This is a breaking change;



### Test Plan

- check automaticDataCollection == false to make sure token refresh is disabled
- see if setting automaticDataCollection == false then setting token refresh programmatically works? (the GDPR scenario)

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter
